suplemon/ui.py

```diff
@@ -240,38 +243,59 @@ def setup_colors(self):
 
         self.app.themes.use(self.app.config["editor"]["theme"])
 
-    def setup_windows(self, resize=False):
-        """Initialize windows."""
-        yx = self.screen.getmaxyx()
+    def setup_windows(self):
+        """Initialize and layout windows."""
+        # We are using curses.newwin instead of self.screen.subwin/derwin because
+        # subwindows are getting a special treatment on resize. e.g., the legend
+        # bar may automatically resize to one line if the window gets smaller.
+        # Even after doing the layout by moving the legend window into its proper
+        # place a call to resize() to restore the 2 line height will error out.
+        # This does not happen with curses.newwin().
+        #
+        # https://anonscm.debian.org/cgit/collab-maint/ncurses.git/tree/ncurses/base/resizeterm.c#n274
+        # https://anonscm.debian.org/cgit/collab-maint/ncurses.git/tree/ncurses/base/wresize.c#n87
         self.text_input = None
-        self.header_win = curses.newwin(1, yx[1], 0, 0)
-        self.status_win = curses.newwin(1, yx[1], yx[0]-1, 0)
-
-        # Test for new curses
-        if "get_wch" not in dir(self.header_win):
-            # Notify only once
-            if not self.warned_old_curses:
```

Comment:
I did not check if self.warned_old_curses has other implications outside of ui.py or can be removed completely after this change. A quick grep -R inside the repo seems to suggest it can be removed completely. If that is indeed the case I'll remove it from `__init__` as well.

Reply:
That's ok. It's not used anywhere else like you said.